swap_meet/vendor.py

```diff
@@ -1,2 +1,156 @@
+                    self.inventory.remove(my_item)
+                    self.inventory.append(their_item)
+                    other_vendor.inventory.append(my_item)
+                    other_vendor.inventory.remove(their_item)
+                    return True
+        return False
+            
+
+    # Wave 4
+    def swap_first_item(self, other_vendor):
+        """ 
+        Swapping the first item of both inventory:
+            - If the first item of both inventories been swapped then return True,
+            - Otherwise, return False.
+        """
+        
+        # checking validation to make sure both inventories are not empty before swapping
+        if len(self.inventory) > 0 and len(other_vendor.inventory) > 0:
+            self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]
+            return True
+        return False
+
+
+    # Wave 6
+    def get_best_by_category(self, category):
+        """ 
+        Findding best item by category:
+            - If it is highest condition and matches category then return that item
+            - Otherwise, return None
+        """
+        
+        # validating the input and finding the highest condition that matches category for best item
+        if category and len(self.inventory) > 0:
+            for item in self.inventory:
+                if (isinstance(item.condition, int) or isinstance(item.condition, float)) and item.category == category :
+                    if item.condition == max(item.condition for item in self.inventory if item.category == category):
+                        return item
+        return None
+
+    
+    def swap_best_by_category(self, other, my_priority, their_priority ):
+        """
+        Swapping by best item:
+            - invoke function get_best_by_category() to find best item
+            - invoked function swap_items() then it return True if item been swapped, otherwise, return False.
+        """
+        
+        # find my best item and other best item base on each priority.
+        my_best_item = self.get_best_by_category(their_priority)
+        other_best_item = other.get_best_by_category(my_priority)
+        # return True or False
+        return self.swap_items(other, my_best_item, other_best_item)
+        
+
+    # Optional Enhancements
+    def get_newest_by_category(self, category):
+        """ 
+        Findding newest item by category:
+            - If it is youngest age and matches category then return that item
+            - Otherwise, return None
+        """
+        
+        # validate input and find min age that matches category for newest item
+        if category and len(self.inventory) > 0:
+            for item in self.inventory:
+                if item.category == category and isinstance(item.age, int):
+                    if item.age == min(item.age for item in self.inventory if item.category == category):
+                        return item
```

Comment:
This works great! As an alternative, can you think of a way to write this that uses a `for` loop, but doesn't use `min`. And as another alternative, can you think of a way to write this that uses `min` but doesn't use a `for` loop?